internal/server/event_list.go

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package server
+
+import (
+	"context"
+	"fmt"
+
+	irievent "github.com/ironcore-dev/ironcore/iri/apis/event/v1alpha1"
+	iri "github.com/ironcore-dev/ironcore/iri/apis/machine/v1alpha1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func (s *Server) filterEvents(events []*irievent.Event, filter *iri.EventFilter) []*irievent.Event {
+	if filter == nil {
+		return events
+	}
+
+	var (
+		res []*irievent.Event
+		sel = labels.SelectorFromSet(filter.LabelSelector)
+	)
+	for _, iriEvent := range events {
+		if !sel.Matches(labels.Set(iriEvent.Spec.InvolvedObjectMeta.Labels)) {
+			continue
+		}
+
+		if filter.EventsFromTime > 0 && filter.EventsToTime > 0 {
+			if iriEvent.Spec.EventTime < filter.EventsFromTime || iriEvent.Spec.EventTime > filter.EventsToTime {
+				continue
+			}
+		}
+
+		res = append(res, iriEvent)
+	}
+	return res
+}
+
+func (s *Server) listEvents() ([]*irievent.Event, error) {
+	events := s.eventStore.ListEvents()
+
+	var iriEvents []*irievent.Event
+	for _, event := range events {
+		iriEvent := &irievent.Event{
+			Spec: &irievent.EventSpec{
+				InvolvedObjectMeta: event.Spec.InvolvedObjectMeta,
+				Reason:             event.Spec.Reason,
+				Message:            event.Spec.Message,
+				Type:               event.Spec.Type,
+				EventTime:          event.Spec.EventTime,
+			},
+		}
+		iriEvents = append(iriEvents, iriEvent)
+	}
+	return iriEvents, nil
+}
+
+func (s *Server) ListEvents(ctx context.Context, req *iri.ListEventsRequest) (*iri.ListEventsResponse, error) {
+	iriEvents, err := s.listEvents()
+	if err != nil {
+		return nil, fmt.Errorf("error listing machine events : %w", err)
```

Comment:
delete space before :

Reply:
I will remove this code since it is no longer needed.

Reply:
removed